pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -852,6 +856,15 @@ func (gce *GCECloud) createTargetPool(name, serviceName, region string, hosts []
 	for _, host := range hosts {
 		instances = append(instances, makeHostURL(gce.projectID, host.Zone, host.Name))
 	}
+	if len(instances) > maxInstancesPerTargetPool {
```

Comment:
Can we always pass in a list of healthy nodes? I don't want someone to have 1000 nodes with dead kube-proxy and 200 others with service endpoints that we can't reach.